packages/flutter/test/material/refresh_indicator_test.dart

```diff
@@ -1203,4 +1203,102 @@ void main() {
       await testColor(color);
     }
   });
+  testWidgets('RefreshIndicator default elevation value correctly',
+    (WidgetTester tester) async {
+  const List<double> positions = <double>[50.0, 100.0, 150.0];
+
+    final AnimationController positionController =
+        AnimationController(vsync: const TestVSync());
+    addTearDown(positionController.dispose);
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: RefreshIndicator(
+          onRefresh: refresh,
+          child: ListView(
+            physics: const AlwaysScrollableScrollPhysics(),
+            children: const <Widget>[Text('X')],
+          ),
+        ),
+      ),
+    );
+
+    RefreshProgressIndicator getIndicator() {
+      return tester.widget<RefreshProgressIndicator>(
+        find.byType(RefreshProgressIndicator),
+      );
+    }
+
+    // Correspond to [_kDragContainerExtentPercentage].
+    final double maxPosition =
+        tester.view.physicalSize.height / tester.view.devicePixelRatio * 0.25;
+    for (final double position in positions) {
+      await tester.fling(find.text('X'), Offset(0.0, position), 1.0);
+      await tester.pump();
+      positionController.value = position / maxPosition;
+      expect(
+        getIndicator().elevation,
+        2.0,
+      );
+      // Wait until the fling finishes before starting the next fling.
+      await tester.pumpAndSettle();
+    }
+  });
+
+  testWidgets('RefreshIndicator custom elevation values correctly',
+      (WidgetTester tester) async {
+    const List<double> elevations = <double>[
+      0,
+      0.1,
+      1,
+      1.1,
+      1.59,
+    ];
+
+    const List<double> positions = <double>[50.0, 100.0, 150.0];
+
+    Future<void> testElevation(double elevation) async {
```

Comment:
The entire `testWidgets` function is meant to test this exact behavior, so, as before, this function is not needed.